src/main/java/at/ac/tuwien/kr/alpha/common/predicates/ExternalEvaluable.java

```diff
@@ -37,7 +38,11 @@ public boolean evaluate(List<Term> terms, Substitution substitution) {
 				it = it.substitute(substitution);
 			}
 
-			arguments[i] = it;
+			if (!(it instanceof ConstantTerm)) {
+				throw new IllegalArgumentException("how did i get here?");
```

Comment:
I guess execution of this branch indicates a bug. Since users might see this first, the message should be more informative like "[what happened]. Should not happen."